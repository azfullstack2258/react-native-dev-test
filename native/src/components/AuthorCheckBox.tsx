import React from 'react';
import { StyleSheet } from 'react-native';
import { CheckBox } from 'react-native-elements';

import { Author } from '../util/types';

interface AuthorCheckBoxProps {
  info: Author;
  isChecked: boolean;
  onPress: (id: string) => void;
}

class AuthorCheckBox extends React.Component<AuthorCheckBoxProps> {
  handleAuthorSelectChange = () => {
    const {
      info: { id },
      onPress,
    } = this.props;
    onPress(id);
  };

  render() {
    const { info, isChecked } = this.props;
    const { name } = info;

    return (
      <CheckBox
        containerStyle={styles.container}
        title={name}
        checked={isChecked}
        onPress={this.handleAuthorSelectChange}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: 'calc(50% - 20px)',
  },
});

export default AuthorCheckBox;
