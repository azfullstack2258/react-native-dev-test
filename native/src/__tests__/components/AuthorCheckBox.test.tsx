import React from 'react';
import renderer from 'react-test-renderer';
import AuthorCheckBox from '../../components/AuthorCheckBox';

const mockProps = {
  info: {
    id: 'mock_author_id_12345678',
    name: 'Test Author',
  },
  isChecked: true,
  onPress: () => {},
};

describe('Author CheckBox', () => {
  it('can render author checkbox', () => {
    const authorCheckBox = renderer
      .create(<AuthorCheckBox {...mockProps} />)
      .toJSON();
    expect(authorCheckBox).toMatchSnapshot();
  });
});
