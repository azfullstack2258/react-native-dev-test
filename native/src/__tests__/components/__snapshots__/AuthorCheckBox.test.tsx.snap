// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author CheckBox can render author checkbox 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
      "padding": 10,
      "width": "calc(50% - 20px)",
    }
  }
  testID="checkbox"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
      testID="checkboxTitle"
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Test Author
    </Text>
  </View>
</View>
`;
